src/agent.rs

```diff
@@ -14,6 +14,21 @@ use {
     cookie_store::CookieStore,
 };
 
+/// Strategy for keeping `authorization` headers during redirects.
+///
+/// `Never` is the default strategy and never preserves `authorization` header in redirects.
+/// `SameHost` send the authorization header in redirects only if the host of the redirect is
+/// the same of the previous request, and both use the `https` scheme.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum RedirectAuthHeaders {
+    /// Never preserve the `authorization` header on redirect. This is the default.
+    Never,
+    /// Preserve the `authorization` header when the redirect is to the same host. Must
+    /// be under the `https` scheme (though port can differ).
```

Comment:
I don't think the `https` scheme restriction makes sense here since we allow setting the authorization header on non-https. Instead, how about: the scheme must be the same. That way if you start on https you can't accidentally carry the authorization header on a redirect to http.

Reply:
Or maybe we allow http -> https but not the reverse?

Reply:
I went with "same or http -> https."